src/components/source_editor/LatticeConfigurationDialog.js

```diff
@@ -108,10 +110,33 @@ class LatticeConfigurationDialog extends React.Component {
         const newMaterial = new Material(newMaterialConfig);
         // assert basis is stored in 'crystal' units
         newMaterial.toCrystal();
+        newMaterial.isNonPeriodic = this.state.isNonPeriodic;
+
         this.props.onUpdate(newMaterial);
         this.props.onSubmit();
     }
 
+    renderNonPeriodicSwitch() {
```

Comment:
Moved the switch into the lattice card, seemed like a better fit to also include an apply switch. 

we could make a separate non-periodic card, but this seemed like the simpler solution